src/operations/command.ts

```diff
@@ -126,6 +127,10 @@ export abstract class CommandOperation<T> extends AbstractOperation<T> {
       Object.assign(cmd, { readConcern: this.readConcern });
     }
 
+    if (this.trySecondaryWrite && serverWireVersion < MIN_SECONDARY_WRITE_WIRE_VERSION) {
```

Comment:
The spec states that if we have a write operation that can execute on a secondary, but the server doesn't satisfy the requirements, there must be no read preference provided. This is backed by the crud spec unified tests.

Reply:
Since this is the only place where "omitReadPreference" is set, is there a benefit to having an `omitReadPreference` defined as a distinct command option as opposed to just performing the option removal right here? (I'm not saying we definitely shouldn't, I just want to understand the reasoning better). And if we do want to keep the option, would `ignoreReadPreference` to be more clear about what we want to make sure happens? (Or is it just that we don't want to send a mismatched preference to, if I understand correctly, an already guaranteed primary server selected?)

Reply:
I used “omit” instead of “ignore” because in many cases there is an explicit read preference set that needs to be removed from the command. Using “ignore” to me would imply no action taken but here we remove it. There is some vagueness going on in the specs, such as https://jira.mongodb.org/plugins/servlet/mobile#issue/DRIVERS-1918 which I figured we would address later.